packages/core-blockchain/lib/blockchain.js

```diff
@@ -528,20 +528,28 @@ module.exports = class Blockchain {
   }
 
   /**
-   * Get unconfirmed transactions for the specified block size.
-   * @param  {Number}  blockSize
-   * @param  {Boolean} forForging
+   * Get the topmost N unconfirmed transactions as of blockchain height `asOfHeight`.
+   * @param {Number} n
+   * @param {Number} asOfHeight
    * @return {Object}
    */
-  getUnconfirmedTransactions(blockSize) {
-    const transactions = this.transactionPool.getTransactionsForForging(
-      blockSize,
-    )
+  async getUnconfirmedTransactions(n, asOfHeight = undefined) {
+    let transactions = this.transactionPool.getTransactions(0, n)
+    const transactionsIds = transactions.map(t => t.id)
+
+    const forgedIds = await this.database.getForgedTransactionsIds(transactionsIds, asOfHeight)
```

Comment:
change to lastForged Height where this forger has succesfully forged. 
`asOfHeight=lastForgedBlockByDelegate.height`